test/fail_compilation/fob2.d

```diff
@@ -0,0 +1,43 @@
+/* Testing Ownership/Borrowing system
+ */
+
+int* malloc();
+void free(int*);
+
+/* TEST_OUTPUT:
+---
+fail_compilation/fob2.d(110): Error: variable `fob2.foo1.b1` no longer has a valid constant copy from `p`
+fail_compilation/fob2.d(103): Error: variable `fob2.foo1.p` is left dangling at return
+---
+*/
+
+#line 100
+
+@live int foo1(int i)
+{
+    int* p = malloc();
+    scope const(int)* b1, b2;
+    if (i)
+        b1 = p;
+    else
+        b2 = p;
+    *p = 3;
+    return *b1;
+}
+
+/* TEST_OUTPUT:
+---
+fail_compilation/fob2.d(203): Error: variable `fob2.zoo2.p` is not Owner, cannot borrow from it
```

Comment:
Is this because of `p + 1`? If so, isn't the error message confusing?

Reply:
No, it's because `p` is sent twice to foo2(). `p + 1` is considered equivalent to `p` by the checker.

Reply:
BTW, I expect error messages can be improved over time. Such polishing, however, won't be in the first iteration.